src/query/sql/src/planner/optimizer/rule/rewrite/rule_push_down_prewhere.rs

```diff
@@ -58,44 +63,65 @@ impl RulePushDownPrewhere {
+                    schema,
+                    columns,
+                }
+            }
+        }
+        impl<'a> Visitor<'a> for ColumnVisitor {
+            fn visit_bound_column_ref(&mut self, column: &'a BoundColumnRef) -> Result<()> {
                 if let Some(index) = &column.column.table_index {
-                    if table_index == *index
+                    if self.table_index == *index
                         && (column.column.visibility == Visibility::InVisible
-                            || schema.index_of(column.column.column_name.as_str()).is_ok())
+                            || self
+                                .schema
+                                .index_of(column.column.column_name.as_str())
+                                .is_ok())
                     {
-                        columns.insert(column.column.index);
+                        self.columns.insert(column.column.index);
                         return Ok(());
                     }
                 }
                 return Err(ErrorCode::Unimplemented("Column is not in the table"));
             }
-            ScalarExpr::FunctionCall(func) => {
-                for arg in func.arguments.iter() {
-                    Self::collect_columns_impl(table_index, schema, arg, columns)?;
-                }
-            }
-            ScalarExpr::CastExpr(cast) => {
-                Self::collect_columns_impl(table_index, schema, cast.argument.as_ref(), columns)?;
+            fn visit_window_function(&mut self, window: &'a WindowFunc) -> Result<()> {
+                return Err(ErrorCode::Unimplemented(format!(
+                    "Prewhere don't support expr {:?}",
+                    window
+                )));
             }
-            ScalarExpr::ConstantExpr(_) => {}
-            ScalarExpr::UDFCall(udf) => {
-                for arg in udf.arguments.iter() {
-                    Self::collect_columns_impl(table_index, schema, arg, columns)?;
-                }
+            fn visit_aggregate_function(&mut self, aggregate: &'a AggregateFunction) -> Result<()> {
+                return Err(ErrorCode::Unimplemented(format!(
+                    "Prewhere don't support expr {:?}",
+                    aggregate
+                )));
             }
-            _ => {
-                // SubqueryExpr and AggregateFunction will not appear in Filter-LogicalGet
+            fn visit_subquery(&mut self, subquery: &'a SubqueryExpr) -> Result<()> {
                 return Err(ErrorCode::Unimplemented(format!(
                     "Prewhere don't support expr {:?}",
-                    expr
+                    subquery
                 )));
             }
         }
-        Ok(())
+
+        let mut column_visitor = ColumnVisitor {
+            table_index,
+            schema: schema.clone(),
+            columns: ColumnSet::new(),
+        };
+        // WindowFunc, SubqueryExpr and AggregateFunction will not appear in Filter-LogicalGet
```

Comment:
Maybe UDFCall can not appear in Filter-LogicalGet? But UDFLambdaCall can?

Reply:
What's `Filter-LogicalGet` ?

Reply:
I think it's Filter logical op? cc @b41sh

Reply:
I think it's Filter logical op? cc @b41sh

Reply:
aggregate and window function can't appear in filter, binder phase will check it.

Reply:
> aggregate and window function can't appear in filter, binder phase will check it.

Yes. https://github.com/datafuselabs/databend/blob/b8f445bd59315d4d80f6306d515e4a11c29e0692/src/query/sql/src/planner/binder/select.rs#L468